clients/pkg/promtail/targets/syslog/syslogtarget.go

```diff
@@ -42,6 +45,7 @@ type SyslogTarget struct {
 
 	messages     chan message
 	messagesDone chan struct{}
+	decoder      *encoding.Decoder
```

Comment:
What do you think about having a `encoding.Encoding` field instead. For default behaviour, you can assign `encoding.Nop` (https://pkg.go.dev/golang.org/x/text@v0.23.0/encoding#Encoding). This would avoid the `nil`-check later in the `messageSender` function.